covid19_sfbayarea/data/alameda/cases_by_gender.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Dict, List
+from .power_bi_querier import PowerBiQuerier
+
+class CasesByGender(PowerBiQuerier):
+    def __init__(self) -> None:
+        self.source = 'v'
+        self.name = 'V_Combined_data'
+        self.property = 'Gender'
+        super().__init__()
+
+    def _parse_data(self, response_json: Dict[str, List]) -> Dict[str, int]:
+        results = super()._parse_data(response_json)
+        data_pairs = filter(lambda result: len(result) == 2, results)
+        return { gender.lower(): count for gender, count in data_pairs }
+
+    def _select(self) -> List[Dict[str, Any]]:
+        property = 'NumberOfCases'
+        return [
+            {
+                'Column': self._column_expression(self.property),
+                'Name': f'{self.name}.{self.property}'
+            },
+            {
+                'Measure': self._column_expression(property),
+                'Name': f'{self.name}.{property}'
+            },
+       ]
```

Comment:
Ahhhhhhhh, I see what’s different! The dashboard’s query has a `Where` section that explicitly selects out `Male`, `Female`, and `Unknown` (also, only rows with a value `> 10`), and we don’t have a `Where` section at all:

```js
{
    "version": "1.0.0",
    "queries": [
        {
            "Query": {
                "Commands": [
                    {
                        "SemanticQueryDataShapeCommand": {
                            "Query": {
                                "Version": 2,
                                "From": [
                                    {
                                        "Name": "v",
                                        "Entity": "V_Combined_data",
                                        "Type": 0
                                    }
                                ],
                                "Select": [
                                    {
                                        "Column": {
                                            "Expression": {
                                                "SourceRef": {
                                                    "Source": "v"
                                                }
                                            },
                                            "Property": "Gender"
                                        },
                                        "Name": "V_Combined_data.Gender"
                                    },
                                    {
                                        "Measure": {
                                            "Expression": {
                                                "SourceRef": {
                                                    "Source": "v"
                                                }
                                            },
                                            "Property": "NumberOfCases"
                                        },
                                        "Name": "V_Combined_data.NumberOfCases"
                                    }
                                ],
                                "Where": [
                                    {
                                        "Condition": {
                                            "In": {
                                                "Expressions": [
                                                    {
                                                        "Column": {
                                                            "Expression": {
                                                                "SourceRef": {
                                                                    "Source": "v"
                                                                }
                                                            },
                                                            "Property": "Gender"
                                                        }
                                                    }
                                                ],
                                                "Values": [
                                                    [
                                                        {
                                                            "Literal": {
                                                                "Value": "'Female'"
                                                            }
                                                        }
                                                    ],
                                                    [
                                                        {
                                                            "Literal": {
                                                                "Value": "'Male'"
                                                            }
                                                        }
                                                    ],
                                                    [
                                                        {
                                                            "Literal": {
                                                                "Value": "'Unknown'"
                                                            }
                                                        }
                                                    ]
                                                ]
                                            }
                                        }
                                    },
                                    {
                                        "Condition": {
                                            "Comparison": {
                                                "ComparisonKind": 2,
                                                "Left": {
                                                    "Measure": {
                                                        "Expression": {
                                                            "SourceRef": {
                                                                "Source": "v"
                                                            }
                                                        },
                                                        "Property": "NumberOfCases"
                                                    }
                                                },
                                                "Right": {
                                                    "Literal": {
                                                        "Value": "10L"
                                                    }
                                                }
                                            }
                                        },
                                        "Target": [
                                            {
                                                "Column": {
                                                    "Expression": {
                                                        "SourceRef": {
                                                            "Source": "v"
                                                        }
                                                    },
                                                    "Property": "Gender"
                                                }
                                            }
                                        ]
                                    }
                                ],
```

Where this *would* go in our code:

https://github.com/sfbrigade/data-covid19-sfbayarea/blob/32f8fc03e6b8ecb5ef680cbecb86f2db8b4675a3/covid19_sfbayarea/data/alameda/power_bi_querier.py#L52-L63

Part of me says we should follow whatever the dashboard is doing, but I worry we won’t know when to change if they ever add other genders. Maybe it’s best just to either:

- Specifically exclude `D` (since it seems like it’s probably bad data).
- Strip out values less than 10 (I think some of our other data sources have that built-in, and the Alameda dashboard seems to be doing it in all the categorizations for cases and for deaths), since those represent potential privacy issues anyway.

Or even do both, I guess. We could include `< 10` in the query but output something in our data so the category is there, but the front-end can display it as “unknown” or “too few” or whatever.

Reply:
I was thinking the `Where` clause was unnecessary as it would only serve to limit the data. I'm going to have a look at stripping these and maybe include the where clause instead as it is what the dashboard is doing. I hadn't thought about privacy issues for small numbers of cases.

Reply:
Heh, I would have made the same assumption until I saw the results! Plus not filtering seems more future-proof (hence why I think it might be better to filter *out* `D` instead of copying the dashboard's query and only asking for specific fields). I admit I’m pretty curious what `D` and the other complex field are, though. 😛

Re: privacy issues, it probably only jumped out at me because one of the other county scrapers used to have special code to handle it (that county returned integers most of the time, but had the string "< 10" when there were too few). Not showing a number for < 10 seems like a good heuristic to follow, but I don't know that we do it explicitly anywhere else — we just count on the fact that the county is only exposing values that have already been filtered. So I think(?) this may be a unique case.